### ISSUE TITLE
schema-ast reorders parts of schema on each generation

### ISSUE THREAD

**Issue by Eddman** [dotansimha/graphql-code-generator#4919]
**Describe the bug**
Our introspection query returns items in random order as the generation is asynchronous and rendered when calculated.
On the `schema-ast` side it just renders in that random order. That way our schema on the client side constantly changes.

With `graphql@14` it was not an issue. With `graphql@15` `schema-ast` does not do any post-processing and reshuffles generated schema.

**To Reproduce**

<!--
#############
#############
🛑 STOP! 🛑 STOP! 🛑 STOP! 🛑 STOP! 🛑 STOP! 🛑 STOP! 🛑 STOP! 🛑 STOP! 🛑 STOP! 🛑 STOP! 🛑 STOP! 

PLEASE READ THE FOLLOWING: 
You can use this template to create a live sandbox of the issue: 
https://codesandbox.io/s/github/dotansimha/graphql-code-generator-issue-sandbox-template
Make sure sure fork this template, and run `yarn generate` in the terminal. 
Please make sure the Codegen and plugins version under `package.json` matches yours. 
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
~~~~~~~~~ ISSUES WITH A LIVE REPRODUCTION WILL BE ANSWERED QUICKLY ~~~~~~~
~~~~~~ ISSUES WITHOUT REPRODUCTION MIGHT GET CLOSED WITHOUT RESPONSE ~~~~~
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
#############
#############
-->

1. My GraphQL schema:

```graphql
N/A
```

2. My GraphQL operations:

```graphql
N/A
```

3. My `codegen.yml` config file:

```yml
schema: https://.../graphql
overwrite: true
generates:
  ./src/schema.graphql:
    plugins:
      - schema-ast
    config:
      includeDirectives: true
      commentDescriptions: true

    # FIXME: remove this hook once https://github.com/dotansimha/graphql-code-generator/issues/4393 has been resolved.
    hooks:
      afterOneFileWrite:
        - sed -i -e ':a' -e 'N' -e '$!ba' -e 's/[ ]*""""""\n//g'
```

**Expected behavior**
Schema is ordered and does not change.

**Environment:**

- OS: Linux
- `@graphql-codegen/schema-ast`: `~1.17.8`
- NodeJS: `v12.16.2`

**Additional context**
<!-- Add any other context about the problem here. -->

**Comment by dotansimha** [comment]
> That way our schema on the client side constantly changes.

The schema doesn't actually change, right? just the order of the items, based on the changes from the introspection. 

> Our introspection query

How do you save the introspection file?

**Comment by Eddman** [comment]
> The schema doesn't actually change, right? just the order of the items, based on the changes from the introspection.

Yes, only the order of generated items is changed.

> How do you save the introspection file?

As you can see in our config: We download it from our endpoint over HTTPS and store it in `./src/schema.graphql` using `schema-ast`. The HTTPS endpoint runs in more replicas behind a loadbalancer in cloud so even result of 2 calls generates differences in order of `./src/schema.graphql`.

**Comment by dotansimha** [comment]
I see. Maybe we can use `lexicographicSortSchema` from `graphlql-js` to fix that, I'll take a look.

**Comment by dotansimha** [comment]
@Eddman implemented in: https://github.com/dotansimha/graphql-code-generator/pull/5034

**Comment by dotansimha** [comment]
Available in `@graphql-codegen/schema-ast@1.18.0`.

**Comment by Eddman** [comment]
@dotansimha Tested and can confirm it works as expected. Thanks!
